core/domain/topic_services.py

```diff
@@ -1856,9 +1857,10 @@ def get_all_published_story_exploration_ids(
                     topic.additional_story_references
                 if story_ref.story_is_published
             ]
-            published_stories: List[story_domain.Story] = [
-                story for story in story_fetchers.get_stories_by_ids(
-                    published_story_ids, strict=False)
+            published_stories_query_result = story_fetchers.get_stories_by_ids(
```

Comment:
Re https://github.com/oppia/oppia/pull/18988#discussion_r1520292614 -- no, this isn't what I meant. The issue I see is that you're making a call to the database for every topic in a for loop, and this results in N calls which increases latency a lot. I'm wondering if you can just make 1 call to the database.